graphix/transpiler.py

```diff
@@ -33,6 +36,104 @@ def __init__(self, width):
         self.width = width
         self.instruction = []
 
+        self.num_parameterized_gates = 0  ## for circuits with parameterized gates
+        self.parameters = set()
+
+    def inspect_parameterized_gates(self, store_data: bool = True, return_data: bool = False):
+        """method to summarize all parameterized gate instructions
+        informs user about all Parameter variables in the circuit
+
+        Parameters
+        ----------
+        store_data : bool
+                    if True saves all information in respective class attribrutes
+        return_data : bool
+                    if True then returns
+                    set - of all the parameters used in the circuit
+
+        """
+
+        if self.num_parameterized_gates == 0:
+            raise ValueError(" no paramaters to assign ")
+
+        elif self.num_parameterized_gates > 0:
```

Comment:
No need to make this part as a condition statement since we know the `num_parameterized_gates` attribute is setted to 0 and we already test it in the if statement above and that it raises an error if so.